frontend/src/components/homepage/Filter.tsx

```diff
@@ -32,39 +34,65 @@ const Filter = ({
   setIsTranslator,
   isDisabled = false,
 }: FilterProps) => {
+  const approvedLanguages = role
+    ? approvedLanguagesTranslation
+    : approvedLanguagesReview;
+
   const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
     setLanguage(event.target.value);
   };
+
   const handleRoleChange = (nextRole: string) => {
+    const newApprovedLanguages =
+      nextRole === "Translator"
+        ? approvedLanguagesTranslation
+        : approvedLanguagesReview;
+    const newLanguage = Object.keys(newApprovedLanguages)[0];
     setIsTranslator(nextRole === "Translator");
+    setLanguage(newLanguage);
   };
+
   const handleLevelChangeStr = (nextLevel: string) => {
     setLevel(parseInt(nextLevel.replace("Level ", ""), 10));
   };
+
   const languageOptions = isDisabled ? (
     <option key="undefined" value="undefined">
       {" "}
     </option>
   ) : (
-    Object.keys(approvedLanguagesTranslation).map((lang) => (
+    Object.keys(approvedLanguages).map((lang) => (
       <option key={lang} value={lang}>
         {convertLanguageTitleCase(lang)}
       </option>
     ))
   );
-  const maxLvl = approvedLanguagesTranslation[language];
-  const lvlOptions = [];
-  for (let i = 1; i <= maxLvl; i += 1) {
-    lvlOptions.push(
-      <option key={i.toString()} value={i.toString()} label={i.toString()} />,
-    );
-  }
+
+  const maxApprovedLevel = approvedLanguages[language];
+  const levelOptions = Array.from(Array(maxApprovedLevel).keys()).map(
```

Comment:
nit: IMO something like this is more clear for a one-liner array generator
```
const arr = [...Array(N+1).keys()].slice(1);
```
policy 71'd it from here https://www.techiedelight.com/create-array-from-1-n-javascript/
What do you think? Lots of interesting options on that site

Reply:
oh yeah I tried using the spread operator and the TS compiler was complaining so I went with `Array.from` instead

i'll try dealing with the ts error to use the spread operator bc I think it looks cleaner too

Reply:
linters do be raining on our parade 😢 feel free to strike a balance between making the linter happy and finding readable code. The current method is not horrendous by any means.